data_steward/cdr_cleaner/cleaning_rules/fix_unmapped_survey_answers.py

```diff
@@ -0,0 +1,198 @@
+"""Original Issues: DC-1043, DC-1053 
+
```

Comment:
I suggest a TL;DR of the problem and how this rule addresses it like:
```suggestion
"""Original Issues: DC-1043, DC-1053 
PPI records may have value_as_concept_id set to 0 by the RDR ETL as a result of changes to the 
vocabulary. This rule sets value_as_concept_id to an appropriate value for these records. Before 
they are updated rows are stored in a sandbox table with the same columns as observation and 
an additional column new_value_as_concept_id.

The RDR ETL populates value_as_concept_id with a concept in the concept_relationship table 
having the 'Maps to value' relationship with value_source_concept_id, however concept_relationship
may not have such a mapping for all PPI answer concepts and in these cases the associated 
value_as_concept_id will get a default value of 0. For these rows this rule uses concepts related 
to value_source_concept_id by 'Maps to' instead. In order to prevent usage of erroneous mappings 
which may be present in the vocabulary, the target concept is also constrained to a limited set 
of concept classes ('Answer', 'Context-dependent', 'Clinical Finding', 'Unit').
```
Arguably some of the comment is TMI as the characterization pertains specifically to the current vocabulary/RDR data. For example I think the numbered bullets are observations - see what I did there? :) - about how concepts end up being mapped currently but these may not always hold true; the update query does not distinguish between standard vs. non-standard or deprecated concepts.

Reply:
Writing has always been my weakness. Your summary is way better than mine, I will accept it blindly.